tools/toolchains/iar.py

```diff
@@ -44,7 +44,7 @@ def check_executable():
         )
 
     def __init__(self, target, notify=None, macros=None, build_profile=None,
-                 build_dir=None):
+                 build_dir=None, coverage_patterns=None):
```

Comment:
As for arm.py; change only applies for gcc-arm so no need to change here.

Reply:
please see my above comments, about why ARM and IAR need to be changed as well.